src/TableauExtract.cc

```diff
@@ -104,7 +105,11 @@ void Extract::HasTable(const FunctionCallbackInfo<Value>& args) {
 }
 
 void Extract::AddTable(const FunctionCallbackInfo<Value>& args) {
-  NodeTde::Table::NewInstanceFromDefinition(args);
+  try {
+    NodeTde::Table::NewInstanceFromDefinition(args);
+  } catch (const Tableau::TableauException& e) {
+    THROW_TABLEAU_EXCEPTION(e);
```

Comment:
A quick test for this would be nice.  Something like...

```javascript
var expect = chai.expect;
// ...
it('throws error when adding invalid table name', function () {
  // Create an extract.
  expectedPath = targetDir + '/mocha-add-invalid-table.tde';
  extract = tableau.dataExtract(expectedPath);

  // Create a table definition.
  tableDef = tableau.tableDefinition();
  tableDef.addColumn('Column', tableau.enums.type('Boolean'));

  // Attempt to add a table with a name other than Extract.
  expect(extract.addTable.bind(extract, 'Invalid', tableDef))
    .to.throw();
});
```